src/Transaction.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Bolt\Thumbs;
+
+use Bolt\Filesystem\Image;
+
+/**
+ * A storage entity for a thumbnail creation request.
+ *
+ * @author Carson Full <carsonfull@gmail.com>
+ */
+class Transaction
+{
+    /** @var string */
+    protected $filePath;
+    /** @var Image */
+    protected $srcImage;
+    /** @var Image */
+    protected $errorImage;
+
+    /** @var string */
+    protected $action;
+    /** @var Dimensions */
+    protected $target;
+
+    /** @var string|null */
+    protected $requestPath;
+
+    /**
+     * Transaction Constructor.
+     *
+     * @param Image|string $file
+     * @param string       $action
+     * @param Dimensions   $dimensions
+     * @param string|null  $requestPath
+     */
+    public function __construct($file, $action = Action::CROP, Dimensions $dimensions = null, $requestPath = null)
+    {
+        if ($file instanceof Image) {
+            $this->srcImage = $file;
+        } else {
+            $this->filePath = $file;
+        }
+        $this->action = $action ?: Action::CROP;
+        $this->target = $dimensions ?: new Dimensions();
+        $this->requestPath = $requestPath;
+    }
```

Comment:
No stress. Someone might try and call it with `new Transaction('foo.jpg', null, null, null)` if they're used to WordP… oh, wait :stuck_out_tongue_winking_eye: